judge/judgedaemon.main.php

```diff
@@ -419,12 +419,15 @@ function fetch_executable_internal(
                         $buildscript .= "javac -cp ./ -d ./ '$source'.java\n";
                         $buildscript .= "echo '#!/bin/sh' > run\n";
                         // no main class detection here
-                        $buildscript .= "echo 'java -cp ./ '$source >> run\n";
+                        $buildscript .= "echo COMPARE_DIR=\$(dirname \"\$0\") >> run\n";
+                        $buildscript .= "echo 'java -cp \"\$COMPARE_DIR\" '$source \"\$@\" >> run\n";
```

Comment:
Why was this not necessary before?

Reply:
Did we ever have a Java compare or run script before?

Reply:
I think but am not 100% sure. Perhaps it also broke after we moved this to the chroot.